homeassistant/components/media_player/epson.py

```diff
@@ -0,0 +1,245 @@
+        DOMAIN, ATTR_CMODE, async_service_handler,
+        schema=EPSON_SCHEMA)
+    return True
+
+
+class EpsonProjector(MediaPlayerDevice):
+    """Representation of Epson Projector Device."""
+
+    def __init__(self, hass, name, host, port, encryption):
+        """Initialize entity to control Epson projector."""
+        self._hass = hass
+        self._name = name
+        import epson_projector as epson
+        from epson_projector.const import DEFAULT_SOURCES
+        self._projector = epson.Projector(
+            host,
+            websession=self._websession(False),
+            port=port)
+
+        self._cmode = None
+        self._source_list = list(DEFAULT_SOURCES.values())
+        self._source = None
+        self._volume = None
+
+        self._state = None
+
+    def _websession(self, verify_ssl):
+        """Return a websession."""
+        return async_get_clientsession(self._hass, verify_ssl)
+
+    async def update(self):
+        """Update state of device."""
+        from epson_projector.const import (
+            EPSON_CODES, POWER,
+            CMODE, CMODE_LIST, SOURCE, VOLUME,
+            BUSY, SOURCE_LIST)
+        is_turned_on = await self._projector.get_property(POWER)
+        _LOGGER.debug("Is turned on %s", is_turned_on)
+        if is_turned_on and is_turned_on == EPSON_CODES[POWER]:
+            self._state = STATE_ON
+            cmode = await self._projector.get_property(CMODE)
+            if cmode and cmode in CMODE_LIST:
+                self._cmode = CMODE_LIST[cmode]
+            source = await self._projector.get_property(SOURCE)
+            if source and source in SOURCE_LIST:
+                self._source = SOURCE_LIST[source]
+            volume = await self._projector.get_property(VOLUME)
+            if volume:
+                self._volume = volume
+        elif is_turned_on == BUSY:
+            self._state = STATE_ON
+        else:
+            self._state = STATE_OFF
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def supported_features(self):
+        """Flag media player features that are supported."""
+        return SUPPORT_EPSON
+
+    async def async_turn_on(self):
+        """Turn on epson."""
+        from epson_projector.const import TURN_ON
+        return await self._projector.send_command(TURN_ON)
+
+    async def async_turn_off(self):
+        """Turn off epson."""
+        from epson_projector.const import TURN_OFF
+        return await self._projector.send_command(TURN_OFF)
+
+    @property
+    def source_list(self):
+        """List of available input sources."""
+        return self._source_list
+
+    @property
+    def source(self):
+        """Get current input sources."""
+        return self._source
+
+    @property
+    def cmode(self):
```

Comment:
This doesn't seem to be used and is not needed. The instance can access the attribute directly.

Reply:
If I delete methods source_list and source, then select source option is not showing up in web ui.

Reply:
Don't remove those. Comments in reviews are directed at the line at the end of the code box shown. In this case that's line 181. So I was only referring to the `cmode` property.